src/libstd/collections/hash/map.rs

```diff
@@ -488,6 +489,57 @@ fn search_hashed_nonempty<K, V, M, F>(table: M, hash: SafeHash, mut is_match: F)
     }
 }
 
+/// Search for a pre-hashed key when the hash map is known to be non-empty.
```

Comment:
This function looks like it may be a copy-pasted version of the one above I think? That's fine but could the comments be removed and instead a comment placed saying it's the ssame one as above just intended for mutable versions?